app/controllers/admin/camps_controller.rb

```diff
@@ -1,36 +1,29 @@
 class Admin::CampsController < Admin::AdminsController
-  include Pagy::Backend
-  
-  before_action :camp_params, only: %i[update create]
-  before_action :set_camp, only: %i[show edit update destroy]
+  before_action :set_camp, only: %i[destroy]
 
   def index
-    @pagy, @camps = pagy(Camp.all)
+    @camps = Camp.all
   end
 
-  def show; end
-
   def new
     @camp = Camp.new
   end
 
   def create
-    @camp = Camp.new(location: params.dig(:camp, :location))
-    if @camp.save
-      redirect_to admin_camp_path(@camp), notice: 'Camp has been created'
+    @camp = Camp.new(camp_params)
+    
+    if @camp.save 
+      redirect_to admin_camps_path, notice: 'Camp has been created'
     else
-      render 'new', alert: @camp.errors.full_messages.to_sentence
+      render 'new', notice: @camp.errors.full_messages.to_sentence
     end
   end
 
-  def edit; end
-
-  def update
-    if @camp.update(location: params.dig(:camp, :location))
-      redirect_to admin_camps_path, notice: 'Camp has been updated'
-    else
-      render 'edit', notice: @camp.errors.full_messages.to_sentence
-    end
+  def update_status
+    @camp = Camp.find(params[:id])
+    @camp_status = @camp.status == "active" ? "inactive" : "active"
```

Comment:
Don't we get status from params which need to be updated? Lets add a method `active?` in Camp model and create constants for status.
```
def active?
  status == ACTIVE
end
```